include/libdnf/rpm/package.hpp

```diff
@@ -38,280 +38,439 @@ class Goal;
 
     /// TODO get_from_repo_id - requires swdb
-    /// For an installed package, return repoid of repo from the package was installed.
+    /// For an installed package, return id of repo from the package was installed.
     /// For an available package, return an empty string.
-    ///
-    /// @replaces dnf:dnf/package.py:attribute:Package.ui_from_repo
-    /// @replaces libdnf:libdnf/dnf-package.h:function:dnf_package_get_origin(DnfPackage * pkg)
+    //
+    // @replaces dnf:dnf/package.py:attribute:Package.ui_from_repo
+    // @replaces libdnf:libdnf/dnf-package.h:function:dnf_package_get_origin(DnfPackage * pkg)
     // void get_from_repo_id() const;
 
-    /// @replaces dnf:dnf/package.py:attribute:Package.hdr_end
-    /// @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_hdr_end(DnfPackage * pkg)
-    unsigned long long get_hdr_end() const;
-
-    /// @replaces dnf:dnf/package.py:attribute:Package.installtime
-    /// @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_installtime(DnfPackage * pkg)
+    /// @return The unix timestamp (seconds since epoch) of when the package was installed.
+    ///         Return 0 if the package is not installed.
+    //
+    // @replaces dnf:dnf/package.py:attribute:Package.installtime
+    // @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_installtime(DnfPackage * pkg)
     unsigned long long get_install_time() const;
 
-    /// @brief Media number for the package
-    ///
-    /// @replaces dnf:dnf/package.py:attribute:Package.medianr
-    /// @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_medianr(DnfPackage * pkg)
-    unsigned long long get_media_number() const;
-
-    /// @brief The rpmdb ID for the package
-    ///
-    /// @replaces dnf:dnf/package.py:attribute:Package.rpmdbid
-    /// @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_rpmdbid(DnfPackage * pkg)
+    /// @return The rpmdb database id (primary key) of the installed RPM package
+    /// @since 5.0
+    //
+    // @replaces dnf:dnf/package.py:attribute:Package.rpmdbid
+    // @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_rpmdbid(DnfPackage * pkg)
     unsigned long long get_rpmdbid() const;
 
-    /// TODO get_repo - requires Repo
-    /// @replaces dnf:dnf/package.py:attribute:Package.repo
-    /// @replaces dnf:dnf/package.py:attribute:Package.repoid
-    /// @replaces dnf:dnf/package.py:attribute:Package.reponame
-    /// @replaces libdnf:libdnf/dnf-package.h:function:dnf_package_get_repo(DnfPackage * pkg)
-    /// @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_reponame(DnfPackage * pkg)
+    /// @return Repo object that represents the repository the package belongs to
+    /// @since 5.0
+    /// @note This isn't the repository the package was installed from.
+    //
+    // @replaces dnf:dnf/package.py:attribute:Package.repo
+    // @replaces libdnf:libdnf/dnf-package.h:function:dnf_package_get_repo(DnfPackage * pkg)
```

Comment:
I forgot what restrictions the bindings impose on this again, but: I think we don't want to return raw pointers on the interface at all if we can avoid it - so this might be slated for changing the return type. In case it's not, the documentation needs to be clear on whether we retain ownership of the pointer or not...

Reply:
Ack. Returning a RepoWeakPtr would be better.
Would you mind creating an issue in our backlog and resolving that in a different PR?
This PR is mainly about docstrings and simple API changes.